crates/core/src/remappings.rs

```diff
@@ -103,33 +104,58 @@ pub fn remappings_foundry(
-            // except the default profile, where we always add the remappings
-            if name == "default" {
-                let new_remappings = generate_remappings(action, paths, soldeer_config, &[])?;
-                let mut array = new_remappings.into_iter().collect::<Array>();
-                format_array(&mut array);
-                profile["remappings"] = value(array);
+    let mut errors = vec![];
+    thread::scope(|scope| {
+        for (name, profile) in profiles.iter_mut() {
+            // we normally only edit remappings of profiles which already have a remappings key
+            let thread = scope.spawn(move || {
+                match profile.get_mut("remappings").map(|v| v.as_array_mut()) {
+                    Some(Some(remappings)) => {
+                        let existing_remappings: Vec<_> = remappings
+                            .iter()
+                            .filter_map(|r| r.as_str())
+                            .filter_map(|r| r.split_once('='))
+                            .collect();
+                        let new_remappings = generate_remappings(
+                            action,
+                            paths,
+                            soldeer_config,
+                            &existing_remappings,
+                        )?;
+                        remappings.clear();
+                        for remapping in new_remappings {
+                            remappings.push(remapping);
+                        }
+                        format_array(remappings);
+                        Ok::<(), RemappingsError>(())
+                    }
+                    _ => {
+                        if name == "default" {
+                            // except the default profile, where we always add the remappings
+                            let new_remappings =
+                                generate_remappings(action, paths, soldeer_config, &[])?;
+                            let mut array = new_remappings.into_iter().collect::<Array>();
+                            format_array(&mut array);
+                            profile["remappings"] = value(array);
+                        }
+
+                        Ok(())
+                    }
+                }
+            });
+            let res = thread.join().expect("Failed to join thread");
+            if res.is_err() {
+                errors.push(res);
             }
-            continue;
-        };
-        let existing_remappings: Vec<_> = remappings
-            .iter()
-            .filter_map(|r| r.as_str())
-            .filter_map(|r| r.split_once('='))
-            .collect();
-        let new_remappings =
-            generate_remappings(action, paths, soldeer_config, &existing_remappings)?;
-        remappings.clear();
-        for remapping in new_remappings {
-            remappings.push(remapping);
         }
-        format_array(remappings);
+    });
```

Comment:
In a huge majority of cases, there will be only one or few profiles. The added complexity and overhead is not justified here.